query-planner-js/src/buildPlan.ts

```diff
@@ -445,6 +449,9 @@ class QueryPlanningTraversal<RV extends Vertex> {
   }
 
   private handleOpenBranch(selection: Selection, options: SimultaneousPathsWithLazyIndirectPaths<RV>[]) {
+    if (performance.now() > this.deadline) {
```

Comment:
We ended up adding deadline checks to some other parts in Rust QP (see [this commit](https://github.com/apollographql/router/pull/6840/commits/adc76b0d6d1e2c94bb35bd38c8497a65276a41cd)), can you also add those deadline checks to the corresponding parts in gateway's QP?